File: TransCoder_tokenized_test_set_functions/java/MAXIMUM_PATH_SUM_POSITION_JUMPS_DIVISIBILITY_CONDITION.java
static void printMaxSum ( int arr [ ], int n ) {
  int dp [ ] = new int [ n ];
  Arrays . fill ( dp, 0 );
  for ( int i = 0;
  i < n;
  i ++ ) {
    dp [ i ] = arr [ i ];
    int maxi = 0;
    for ( int j = 1;
    j <= Math . sqrt ( i + 1 );
    j ++ ) {
      if ( ( ( i + 1 ) % j == 0 ) && ( i + 1 ) != j ) {
        if ( dp [ j - 1 ] > maxi ) maxi = dp [ j - 1 ];
        if ( dp [ ( i + 1 ) / j - 1 ] > maxi && j != 1 ) maxi = dp [ ( i + 1 ) / j - 1 ];
      }
    }
    dp [ i ] += maxi;
  }
  for ( int i = 0;
  i < n;
  i ++ ) System . out . print ( dp [ i ] + " " );
}

File: TransCoder_tokenized_test_set_functions/python/MAXIMUM_PATH_SUM_POSITION_JUMPS_DIVISIBILITY_CONDITION.py
def printMaxSum ( arr, n ) :
    dp = [ 0 for i in range ( n ) ]
    for i in range ( n ) :
        dp [ i ] = arr [ i ]
        maxi = 0
        for j in range ( 1, int ( ( i + 1 ) * * 0.5 ) + 1 ) :
            if ( ( i + 1 ) % j == 0 and ( i + 1 ) != j ) :
                if ( dp [ j - 1 ] > maxi ) :
                    maxi = dp [ j - 1 ]
                if ( dp [ ( i + 1 ) //j-1]>maxiandj!=1):
                    maxi=dp[(i+1)//j-1]
        dp[i]+=maxi
    foriinrange(n):
        print(dp[i],end=' ')
    
